src/main/java/com/vinyl/service/UserService.java

```diff
@@ -35,10 +35,10 @@ public User addUser(User user) {
 		user.setAddress("Defaul Address");
 		user.setRole(r);
 
-		Validator nameValidator = accountValidator.getValidator("NAME_VALIDATOR");
+		Validator<User> nameValidator = accountValidator.getUserNameValidator();
```

Comment:
inline the call to validate(), the local variable won't be needed then, same for the other call